main.cpp

```diff
@@ -37,11 +38,20 @@
 #include <windows.h> /// for getting PID, for proper RNG for MinGW
 #endif
 
+volatile sig_atomic_t userExitFlag = 0;
+void catchCtrlC(int signalID) {
+  signal(SIGINT, catchCtrlC);
```

Comment:
Why are you posting the signal signature here? It's already being posted in main, and this function would not get called  if the signal were not posted. So line 43 is redundant.

Reply:
Actually, the signal is deregistered when caught every time, so I register it again to post a warning / confirmation when it's caught a second time during force exit. But good catch, because I never print that warning! The change has now been made.

Reply:
printf is used because it is thead-agnostic and forces output to the screen, at least much better than the c++ way in my experience, which guarantees nothing of execution/output ordering.

Reply:
I don't understand the "signal is deregistered" bit. If I comment out line 47, The catchCtrlC function still gets called on the second ctrl-c. Is this a platform-specific thing? Can you point me to the reference on this?

Reply:
True. I remember why I did that then, and it shouldn't be there now.